bigtable/client/internal/endian.h

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_INTERNAL_ENDIAN_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_INTERNAL_ENDIAN_H_
+
+#include <bitset>
+#include <limits>
+#include <vector>
+
+#include "bigtable/client/internal/strong_type.h"
+#include "bigtable/client/internal/throw_delegate.h"
+
+#ifdef _MSC_VER
+#include <stdlib.h>
+#elif defined(__GNUC__) || defined(__clang__)
+#include <byteswap.h>
+#endif
+
+#include <string>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+using bigendian64_t = internal::StrongType<std::int64_t, struct BigEndianType>;
+
+namespace internal {
+
+#ifndef _MSC_VER
```

Comment:
Why not:

```C++
#if !defined(_MSC_VER) && !defined(__GNUC__) && !defined(__clang__)
```

Or better yet, since this function is used only once, move the code into the right branch of the `#if/#elif` block below (~ line 64).